File: solutions/javascript/lasagna/1/lasagna.js
// Task 1: Define the expected oven time
export const EXPECTED_MINUTES_IN_OVEN = 40;

// Task 2: Calculate remaining oven time
export function remainingMinutesInOven(actualMinutesInOven) {
  return EXPECTED_MINUTES_IN_OVEN - actualMinutesInOven;
}

// Task 3: Calculate preparation time
export function preparationTimeInMinutes(numberOfLayers) {
  return numberOfLayers * 2;
}

// Task 4: Calculate total working time
export function totalTimeInMinutes(numberOfLayers, actualMinutesInOven) {
  return preparationTimeInMinutes(numberOfLayers) + actualMinutesInOven;
}

File: solutions/r/lasagna/1/lasagna.R
# Define the expected oven time in minutes
expected_minutes_in_oven <- function() {
  60
}

# Calculate the remaining oven time in minutes
remaining_time_in_minutes <- function(actual_minutes) {
  expected_minutes_in_oven() - actual_minutes
}

# Calculate the preparation time in minutes
prep_time_in_minutes <- function(number_of_layers) {
  number_of_layers * 2
}

# Calculate the elapsed time in minutes
elapsed_time_in_minutes <- function(number_of_layers, actual_minutes_in_oven) {
  prep_time_in_minutes(number_of_layers) + actual_minutes_in_oven
}